src/build/utils.js

```diff
@@ -151,6 +151,32 @@ exports.dockerRun = async ({baseDir, logfile, command, env, binds, workingDir, i
   }
 };
 
+// Decode the multiplexed stream from Docker
+// https://docs.docker.com/engine/api/v1.37/#operation/ContainerRename
+class DemuxDockerStream extends Transform {
+  constructor() {
+    super();
+    this.buffer = Buffer.alloc(0);
+  }
+
+  _transform(chunk, encoding, callback) {
+    if (this.buffer.length) {
+      this.buffer = Buffer.concat(this.buffer, chunk);
+    } else {
+      this.buffer = chunk;
+    }
+    while (this.buffer.length >= 8) {
+      // type is at byte 0, but ignored here as we do the same with both
+      // stderr and stdout.
+      const len = this.buffer.readUInt32BE(4);
+      const payload = this.buffer.slice(8, len + 8);
+      this.buffer = this.buffer.slice(len + 8);
```

Comment:
I have a Q about lines `171-173` <--- how does it work?

Reply:
The link a few lines above has the context.  Docker outputs data in packets with an 8-byte header:
[TYPE, 0, 0, 0, SIZE1, SIZE2, SIZE3, SIZE4, ...data].  So this reads the size, then reads exactly that many bytes and sends them to the stream's output.  We ignore the type because it just distinguishes stderr from stdout and we want to mix the two.

Reply:
You mean `https://docs.docker.com/engine/api/v1.37/#operation/ContainerRename`? I looked at it, but it still wasn't clear to me 😞 (maybe I didn't click something there)

Thank you, I got it now! I wonder if your explanation could be added in the comment?